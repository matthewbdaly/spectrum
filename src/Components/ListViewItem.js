// @flow
import React, { Component } from 'react';
import type { Node } from 'react';
import './ListViewItem.scss';

type Props = {
  children?: Node
}

class ListViewItem extends Component<Props> {
  render() {
    return (
      <li className="listview-item">
        {this.props.children}
      </li>
    );
  }
}

export default ListViewItem;
