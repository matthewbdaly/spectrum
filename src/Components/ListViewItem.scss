@import "../variables";

.listview-item {
    width: 100%;
    border: 1px solid lightgray;
    padding: 20px;
    box-sizing: border-box;
}
