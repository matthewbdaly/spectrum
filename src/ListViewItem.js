import React, { Component } from 'react';
import './ListViewItem.scss';

class ListViewItem extends Component {
  render() {
    return (
      <li className="listview-item">
        {this.props.children}
      </li>
    );
  }
}

export default ListViewItem;
