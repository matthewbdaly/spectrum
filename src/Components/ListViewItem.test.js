import React from 'react';
import ReactDOM from 'react-dom';
import ListViewItem from './ListViewItem';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ListViewItem />, div);
});
