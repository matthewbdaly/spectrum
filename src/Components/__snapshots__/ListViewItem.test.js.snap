// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
ListViewItem {
  "_reactInternalFiber": FiberNode {
    "_debugID": 7,
    "_debugIsCurrentlyTiming": false,
    "_debugOwner": null,
    "_debugSource": Object {
      "fileName": "<PROJECT_ROOT>/src/Components/ListViewItem.test.js",
      "lineNumber": 12,
    },
    "alternate": null,
    "child": FiberNode {
      "_debugID": 8,
      "_debugIsCurrentlyTiming": false,
      "_debugOwner": [Circular],
      "_debugSource": Object {
        "fileName": "<PROJECT_ROOT>/src/Components/ListViewItem.js",
        "lineNumber": 13,
      },
      "alternate": null,
      "child": null,
      "effectTag": 0,
      "expirationTime": 0,
      "firstEffect": null,
      "index": 0,
      "internalContextTag": 0,
      "key": null,
      "lastEffect": null,
      "memoizedProps": Object {
        "children": undefined,
        "className": "listview-item",
      },
      "memoizedState": null,
      "nextEffect": null,
      "pendingProps": null,
      "ref": null,
      "return": [Circular],
      "sibling": null,
      "stateNode": <li
        class="listview-item"
/>,
      "tag": 5,
      "type": "li",
      "updateQueue": null,
    },
    "effectTag": 1,
    "expirationTime": 0,
    "firstEffect": null,
    "index": 0,
    "internalContextTag": 0,
    "key": null,
    "lastEffect": null,
    "memoizedProps": Object {},
    "memoizedState": null,
    "nextEffect": null,
    "pendingProps": null,
    "ref": null,
    "return": FiberNode {
      "_debugID": 5,
      "_debugIsCurrentlyTiming": false,
      "_debugOwner": null,
      "_debugSource": null,
      "alternate": FiberNode {
        "_debugID": 5,
        "_debugIsCurrentlyTiming": false,
        "_debugOwner": null,
        "_debugSource": null,
        "alternate": [Circular],
        "child": null,
        "effectTag": 0,
        "expirationTime": 1,
        "firstEffect": null,
        "index": 0,
        "internalContextTag": 0,
        "key": null,
        "lastEffect": null,
        "memoizedProps": null,
        "memoizedState": null,
        "nextEffect": null,
        "pendingProps": null,
        "ref": null,
        "return": null,
        "sibling": null,
        "stateNode": Object {
          "containerInfo": <div>
            <li
                        class="listview-item"
            />
</div>,
          "context": Object {},
          "current": [Circular],
          "finishedWork": null,
          "hydrate": false,
          "isReadyForCommit": false,
          "nextScheduledRoot": null,
          "pendingChildren": null,
          "pendingContext": null,
          "remainingExpirationTime": 0,
        },
        "tag": 3,
        "type": null,
        "updateQueue": Object {
          "baseState": null,
          "callbackList": null,
          "expirationTime": 1,
          "first": Object {
            "callback": null,
            "expirationTime": 1,
            "isForced": false,
            "isReplace": false,
            "next": null,
            "nextCallback": null,
            "partialState": Object {
              "element": <ListViewItem />,
            },
          },
          "hasForceUpdate": false,
          "isInitialized": false,
          "isProcessing": false,
          "last": Object {
            "callback": null,
            "expirationTime": 1,
            "isForced": false,
            "isReplace": false,
            "next": null,
            "nextCallback": null,
            "partialState": Object {
              "element": <ListViewItem />,
            },
          },
        },
      },
      "child": [Circular],
      "effectTag": 0,
      "expirationTime": 0,
      "firstEffect": [Circular],
      "index": 0,
      "internalContextTag": 0,
      "key": null,
      "lastEffect": [Circular],
      "memoizedProps": null,
      "memoizedState": Object {
        "element": <ListViewItem />,
      },
      "nextEffect": null,
      "pendingProps": null,
      "ref": null,
      "return": null,
      "sibling": null,
      "stateNode": Object {
        "containerInfo": <div>
          <li
                    class="listview-item"
          />
</div>,
        "context": Object {},
        "current": [Circular],
        "finishedWork": null,
        "hydrate": false,
        "isReadyForCommit": false,
        "nextScheduledRoot": null,
        "pendingChildren": null,
        "pendingContext": null,
        "remainingExpirationTime": 0,
      },
      "tag": 3,
      "type": null,
      "updateQueue": null,
    },
    "sibling": null,
    "stateNode": [Circular],
    "tag": 2,
    "type": [Function],
    "updateQueue": null,
  },
  "_reactInternalInstance": Object {},
  "context": Object {},
  "props": Object {},
  "refs": Object {},
  "state": null,
  "updater": Object {
    "enqueueForceUpdate": [Function],
    "enqueueReplaceState": [Function],
    "enqueueSetState": [Function],
    "isMounted": [Function],
  },
}
`;
